core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1793,10 +1793,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * has overridden the call site using `setCallSite()`, this will return the user's version.
    */
   private[spark] def getCallSite(): CallSite = {
-    Option(getLocalProperty(CallSite.SHORT_FORM)).map { case shortCallSite =>
-      val longCallSite = Option(getLocalProperty(CallSite.LONG_FORM)).getOrElse("")
-      CallSite(shortCallSite, longCallSite)
-    }.getOrElse(Utils.getCallSite())
+    val callSite = Utils.getCallSite()
+    CallSite(
```

Comment:
Earlier the longform callsite property was not respected (that is it would not be returned even if it was set) unless the shortform callsite is set. That is not intuitive. This change actually simplifies the logic - both short form and long form are respected independent of whether the other one is set.